library/agent/ServiceConfig.ts

```diff
@@ -126,6 +136,42 @@ export class ServiceConfig {
     return { blocked: false };
   }
 
+  private setAllowedIPAddresses(ipAddresses: IPList[]) {
+    this.allowedIPAddresses = [];
+
+    for (const source of ipAddresses) {
+      // Skip empty allowlists
+      if (source.ips.length === 0) {
+        continue;
+      }
+      this.allowedIPAddresses.push({
+        allowlist: new IPMatcher(source.ips),
+        description: source.description,
+      });
+    }
+  }
+
+  updateAllowedIPAddresses(ipAddresses: IPList[]) {
+    this.setAllowedIPAddresses(ipAddresses);
+  }
+
+  isAllowedIPAddress(ip: string): { allowed: boolean } {
```

Comment:
How do we deal with invalid IP addresses here?

Reply:
Invalid IPs in the Geoip list are simply ignored.